### DISCUSSION TITLE
Ethernet - client.connect connects to host every second call

### DISCUSSION THREAD

**Discussion by xRadx** [stm32duino/Arduino_Core_STM32#2266] [Libraries] [answered]
with the code as follow
```
if (TimeNow > nextDataPoll)
{
  nextDataPoll = TimeNow + 10000;
  Serial.println(client.connected());
  if (client.connect("my.host.xx", 23) )
  {
    String msg = "o111111";
    client.print(msg);
   }
  client.stop();
}
```
it seems that client.connect connects to a host every second call (or rather client.stop disconnects) every second call.

Serial monitor:
20:28:12.985 -> 0
20:28:23.004 -> 0
20:28:33.009 -> 1
20:28:42.982 -> 0
20:28:52.993 -> 1

At the first client.connect and then only when "1", the message reaches a host.

I am using Nucleo F746ZG board.

What's the problem/solution?

**Comment by xRadx** [comment]
not using client.stop is a sort of "quick fix" ...but it's not the way it should be.

**Reply by fpistm** [reply, marked as answer]
Hi @xRadx 
Well, my guess is that opening, closing take more time you think. The LwIP set the cocket at `TCP_CLOSING `state and it does not close immediately.
There some other mechanism in LwIP using the `TCP_WAIT` and other.
So you have to take in account.... having a connect each 10s is not really usefull... simply stay connected.

**Reply by xRadx** [reply]
Once I have some free time then I will make a test how long it takes to close a socket, if it always closes and compare with some Wizznet modules.
